4.1rc2: test_generate_config fails when astropy installed on different disk from test run

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

### Description
<!-- Provide a general description of the bug. -->

```
____________________________ test_generate_config _____________________________

tmp_path = WindowsPath('C:/.../pytest-1/test_generate_config0')

    def test_generate_config(tmp_path):
        from astropy.config.configuration import generate_config
        out = io.StringIO()
>       generate_config('astropy', out)

C:\...\astropy\config\tests\test_configs.py:135:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\...\astropy\config\configuration.py:625: in generate_config
    importlib.import_module(mod.name)
C:\...\importlib\__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:783: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
C:\...\astropy\convolution\setup_package.py:9: in <module>
    C_CONVOLVE_PKGDIR = os.path.relpath(os.path.dirname(__file__))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

path = 'C:\\...\\astropy\\convolution'
start = '.'

    def relpath(path, start=None):
        """Return a relative version of a path"""
        path = os.fspath(path)
        if isinstance(path, bytes):
            sep = b'\\'
            curdir = b'.'
            pardir = b'..'
        else:
            sep = '\\'
            curdir = '.'
            pardir = '..'

        if start is None:
            start = curdir

        if not path:
            raise ValueError("no path specified")

        start = os.fspath(start)
        try:
            start_abs = abspath(normpath(start))
            path_abs = abspath(normpath(path))
            start_drive, start_rest = splitdrive(start_abs)
            path_drive, path_rest = splitdrive(path_abs)
            if normcase(start_drive) != normcase(path_drive):
>               raise ValueError("path is on mount %r, start on mount %r" % (
                    path_drive, start_drive))
E                   ValueError: path is on mount 'C:', start on mount 'D:'

C:\...\lib\ntpath.py:703: ValueError
```

--- a/astropy/convolution/setup_package.py
+++ b/astropy/convolution/setup_package.py
@@ -6,7 +6,7 @@ from setuptools import Extension
 
 import numpy
 
-C_CONVOLVE_PKGDIR = os.path.relpath(os.path.dirname(__file__))
+C_CONVOLVE_PKGDIR = os.path.abspath(os.path.dirname(__file__))
 
 SRC_FILES = [os.path.join(C_CONVOLVE_PKGDIR, filename)
              for filename in ['src/convolve.c']]

--- a/astropy/io/ascii/setup_package.py
+++ b/astropy/io/ascii/setup_package.py
@@ -5,7 +5,7 @@ from setuptools import Extension
 
 import numpy
 
-ROOT = os.path.relpath(os.path.dirname(__file__))
+ROOT = os.path.abspath(os.path.dirname(__file__))
 
 
 def get_extensions():

--- a/astropy/modeling/setup_package.py
+++ b/astropy/modeling/setup_package.py
@@ -11,7 +11,7 @@ from extension_helpers import import_file
 wcs_setup_package = import_file(join('astropy', 'wcs', 'setup_package.py'))
 
 
-MODELING_ROOT = os.path.relpath(os.path.dirname(__file__))
+MODELING_ROOT = os.path.abspath(os.path.dirname(__file__))
 MODELING_SRC = join(MODELING_ROOT, 'src')
 SRC_FILES = [join(MODELING_SRC, 'projections.c.templ'),
              __file__]

--- a/astropy/table/setup_package.py
+++ b/astropy/table/setup_package.py
@@ -5,7 +5,7 @@ from setuptools import Extension
 
 import numpy
 
-ROOT = os.path.relpath(os.path.dirname(__file__))
+ROOT = os.path.abspath(os.path.dirname(__file__))
 
 
 def get_extensions():

--- a/astropy/timeseries/periodograms/bls/setup_package.py
+++ b/astropy/timeseries/periodograms/bls/setup_package.py
@@ -7,7 +7,7 @@ from setuptools import Extension
 
 import numpy
 
-BLS_ROOT = os.path.relpath(os.path.dirname(__file__))
+BLS_ROOT = os.path.abspath(os.path.dirname(__file__))
 
 
 def get_extensions():

--- a/astropy/utils/setup_package.py
+++ b/astropy/utils/setup_package.py
@@ -1,7 +1,7 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 from setuptools import Extension
-from os.path import dirname, join, relpath
+from os.path import dirname, join, abspath
 
 ASTROPY_UTILS_ROOT = dirname(__file__)
 
@@ -9,5 +9,5 @@ ASTROPY_UTILS_ROOT = dirname(__file__)
 def get_extensions():
     return [
         Extension('astropy.utils._compiler',
-                  [relpath(join(ASTROPY_UTILS_ROOT, 'src', 'compiler.c'))])
+                  [abspath(join(ASTROPY_UTILS_ROOT, 'src', 'compiler.c'))])
     ]

--- a/astropy/wcs/setup_package.py
+++ b/astropy/wcs/setup_package.py
@@ -15,7 +15,7 @@ import numpy
 
 from extension_helpers import import_file, write_if_different, get_compiler, pkg_config
 
-WCSROOT = os.path.relpath(os.path.dirname(__file__))
+WCSROOT = os.path.abspath(os.path.dirname(__file__))
 WCSVERSION = "7.3.0"
 
 

